File: app/src/main/java/com/example/anton/sprintcalendar/Team.java
package com.example.anton.sprintcalendar;

import java.util.Arrays;
import java.util.Collections;
import java.util.Iterator;
import java.util.List;

public class Team implements Iterable<TeamMember> {
    public static final TeamMember TEAM_MEMBER_PLACEHOLDER = new TeamMember("");
    private List<TeamMember> teamMembers;

    public Team(TeamMember... teamMembers) {
        this.teamMembers = Arrays.asList(teamMembers);
    }


    public Iterator<TeamMember> iterator() {
        return new Iterator<TeamMember>() {
            Iterator<TeamMember> delegate = teamMembers.iterator();

            @Override
            public boolean hasNext() {
                return delegate.hasNext();
            }

            @Override
            public TeamMember next() {
                return delegate.next();
            }

            @Override
            public void remove() {

            }
        };
    }

    public TeamMember member(int index) {
        if (index < teamMembers.size()) {
            return teamMembers.get(index);
        } else {
            return TEAM_MEMBER_PLACEHOLDER;
        }
    }

    public int getMembersCount() {
        return teamMembers.size();
    }
}

File: app/src/main/kotlin/net/sourcefusion/agiletools/sprintcalendar/Team.kt
package net.sourcefusion.agiletools.sprintcalendar

import java.util.*

class Team(teamMembers: List<TeamMember>) : Iterable<TeamMember> {
    constructor(vararg teamMembers: TeamMember): this(listOf(*teamMembers))
    private val teamMembers: MutableList<TeamMember>

    init {
        this.teamMembers = ArrayList(teamMembers)
    }

    fun addTeamMember(name: String):TeamMember {
        val teamMember = TeamMember(name)
        teamMembers.add(teamMember)
        return teamMember
    }

    override fun iterator(): Iterator<TeamMember> {
        return object : Iterator<TeamMember> {
            internal var delegate = if(teamMembers.size>0) teamMembers.iterator() else arrayOf(Team.TEAM_MEMBER_PLACEHOLDER).iterator()

            override fun hasNext(): Boolean {
                return delegate.hasNext()
            }

            override fun next(): TeamMember {
                return delegate.next()
            }
        }
    }

    fun member(index: Int): TeamMember {
        if (index < teamMembers.size) {
            return teamMembers[index]
        } else {
            return TEAM_MEMBER_PLACEHOLDER
        }
    }

    val membersCount: Int
        get() = teamMembers.size

    companion object {
        val TEAM_MEMBER_PLACEHOLDER = TeamMember("")
    }
}
